### PR TITLE
feat: Serve run

### PR DESCRIPTION
**Pull request by GGXXLL**
## feature
- srvgrpc
  + Add healthcheck module
  + Add metrics  module
- serve
  + Split service code block
  + ~~If there is no service to apply，then service will not run. The `disable` option is still available.~~
  + Add more debug log information when serve run

### CODE REVIEW

**Comment by Reasno** [comment]
> "If there is no service to apply，then service will not run. The disable option is still available."

I have considered this, but:

1. this is a major breaking change both in compile-time and runtime.
2. An void ApplyRouter call doesn't mean the router has no route, neither are other collectors. For example, I can add routes outside of the framework, and use DI to substitute the default router. 

The rest changes seem pretty good to me.

**Review by Reasno — DISMISSED** [review]

**Comment by Reasno** [comment]
@GGXXLL 这个测试为什么把disable http去掉了

**Review by nfangxu — DISMISSED** [review]

**Comment by GGXXLL** [comment]
> @GGXXLL 这个测试为什么把disable http去掉了。 并行跑的时候 都监听8080 有一定几率冲突。

> @GGXXLL 这个测试为什么把disable http去掉了。 并行跑的时候 都监听8080 有一定几率冲突。

恩恩， 这个没注意到。

感觉这个测试没有必要了， 它跟`TestC_Serve`相比，只是少了module的提供。

File: contract/container.go [hunk 1]
```diff
@@ -11,12 +11,12 @@ import (
 
 // Container holds modules.
 type Container interface {
-	ApplyRouter(router *mux.Router)
-	ApplyGRPCServer(server *grpc.Server)
+	ApplyRouter(router *mux.Router) int
```
[the remark this answers is not in the document]
Comment (GGXXLL) [reply contract/container.go]: LGTM,  这个功能不合适，已经撤销了相应的修改

**Review by nfangxu — APPROVED** [review]

**Pull request merged**